adr/0017-zarf-bundle.md

```diff
@@ -0,0 +1,93 @@
+# 17. Bundles
+
+Date: 2023-06-13
+
+## Status
+
+Pending
+
+## Context
+
+Orchestrating Zarf packages into meta-packages is a current weakpoint of Zarf. The core of Zarf was oriented around components as capabilities, but as Zarf packages have scaled, it has been discovered that entire packages are now rapidly becoming modularized capabilities.
+
+Currently there is no official way to enable the deployment, publishing, pulling, and creation of these Zarf packages. Due to this shortfalling, it has led to the community developing such antipatterns as:
+
+```yaml
+- name: init
+  required: true
+  files:
+    - source: zarf-init-amd64-v0.27.0.tar.zst
+      target: zarf-init-amd64-v0.27.0.tar.zst
+  actions:
+    onDeploy:
+      after:
+        - cmd: zarf package deploy zarf-init-amd64-v0.27.0.tar.zst --components git-server --confirm -l warn
+```
+
+While this _does_ fulfill the need to deploy two packages in one command, it does so in such a way that is very brittle and creates bloated and inefficient packages.
+
+### Proposed Solutions
+
+There are currently three proposed solutions to this, each with their own pros and cons:
+
+#### Zarf Bundle
+
+Uses OCI and a separate zarf schema / declarative YAML definition to pull packages into a single artifact and orchestrate them together.
+
+Biggest problem to solve is how to handle variables and whether/how those should be shared between packages.
+
+Package sources in bundles would be OCI _only_, and would not support local packages.
+
+#### Super Zarf Packages
+
+Add a packages key or another way to overlay packages into a larger package with the same internal structure.
+
+Biggest problem here is how we would scope things after layering things together such as how we might need to change Helm chart name generation or other package level things (like deployed package secrets)
```

Comment:
```suggestion
Adds a packages key or another way to overlay packages into a larger package with the same internal structure as current Zarf packages.

Pros:
- packages would maintain the same syntax under `zarf package` between normal / meta packages.

Cons:
- it would be difficult to properly scope things like variables and helm chart names properly across packages.
- this continues to add to `zarf package` making it more complex and harder to test
```